### ISSUE TITLE
[BUG] macOS: legacy 'setup.py install' installs are broken in 63.1.0 with Python 3.9

### ISSUE THREAD

**Issue by ghost** [pypa/setuptools#3431]
### setuptools version

setuptools==63.1.0

### Python version

Python 3.9.x

### OS

macOS 10.15 or 11 under x86_64 (may happen in other versions as well)

### Additional environment information

This doesn't happen if we do have the `wheel` package installed, nor under setuptools 63.0.0. I didn't see this happening on other versions of Python nor other OSs either.

### Description

I tried to install `crcmod` using setuptools 63.1.0 without the `wheel` package installed. This worked on 63.0.0, but on 63.1.0 it looks like Python isn't able to find the module.

### Expected behavior

Python should be able to find the newly-installed module.

### How to Reproduce

1. Make sure you don't have the `wheel` package installed:
```
python3.9 -m pip uninstall wheel
```

2. Install a package which will trigger a legacy setup.py install if `wheel` isn't found. For example:
```
python3.9 -m pip install crcmod
```

You should see a warning about `Using legacy 'setup.py install' for crcmod, since package 'wheel' is not installed.`

3. Try to use the newly-installed package:

```
python3.9 -c "import crcmod"
```

### Output

```console
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'crcmod'
```

**Comment by A6GibKm** [comment]
I don't know if it is related, but I am also having issues in Linux with libpwquality which also doesn't build wheels, see https://gitlab.gnome.org/GNOME/gnome-build-meta/-/issues/516.

**Comment by nanonyme** [comment]
Note that the libpwquality case is worse in the sense that it will explicitly call setup.py install through makefile. It cannot be worked around simply through installing wheel.

**Comment by nanonyme** [comment]
After reading into this deeper, these two cases may not be related. Pip invokes setuptools with `--single-version-externally-managed` which triggers yet another codepath in setuptools (there are at least three ways for package to get installed). If setup.py is invoked without this parameter, end result is that egg will get intalled instead of a flat directory.

**Comment by nanonyme** [comment]
@martingalvan-nordic if you would like to run pip install with -v (for verbose), you could see what exact parameters it passes to setuptools.

**Comment by martingalvan-nordic** [comment]
@nanonyme regardless of what pip does internally, the bug is still there at least for me. I believe debugging this is a task for the setuptools maintainers.

**Comment by nanonyme** [comment]
Well, it looks like there is API between pip and setuptools. With matching version ranges of pip and setuptools you get sensible results.

**Comment by martingalvan-nordic** [comment]
Yes, but such internal details shouldn't leak to normal users. If pip stops being compatible with setuptools at some version ranges, IMHO this should be very explicitly documented somewhere.
